test/file/common.c

```diff
@@ -68,3 +68,36 @@ void file_tear_down(FILE *fp, const char *funcname)
   if (res == EOF)
     TEST_MSG("Failed to tear down test %s", funcname);
 }
+
+const char *HomeDir;
+
+static const char *get_test_path(void)
+{
+  static const char *path = NULL;
+  if (!path)
+    path = mutt_str_getenv("NEOMUTT_TEST_DIR");
+  if (!path)
+    path = "";
+
+  return path;
+}
+
+void test_gen_path(char *buf, size_t buflen, const char *fmt)
+{
+  snprintf(buf, buflen, NONULL(fmt), get_test_path());
+}
+
+void test_gen_dir(char *buf, size_t buflen, const char *fmt)
+{
+  static const char *dir = NULL;
+  if (!dir)
+  {
+    const char *path = get_test_path();
+    const char *slash = strrchr(path, '/');
+    dir = slash + 1;
+  }
+  if (!dir)
```

Comment:
I guess `dir` will never be NULL here, at best NULL + 1.

Reply:
True.  I'm missing an `if (slash)` before the `dir = `.
However, the function isn't used any more, so it goes.